src/paint.js

```diff
@@ -198,4 +198,50 @@ export default function paint($element, layout) {
   $('#' + id + ' rect.range.s2').attr('fill', 'rgb(' + Math.floor(rangeRGB.r * middleRangeThresh) + ', ' + Math.floor(rangeRGB.g * middleRangeThresh) + ', ' + Math.floor(rangeRGB.b * middleRangeThresh) + ')');
   $('#' + id + ' rect.range.s1').attr('fill', 'rgb(' + Math.floor(rangeRGB.r * lowerRangeThresh) + ', ' + Math.floor(rangeRGB.g * lowerRangeThresh) + ', ' + Math.floor(rangeRGB.b * lowerRangeThresh) + ')');
   $('#' + id + ' rect.range.s0').attr('fill', 'rgb(' + rangeRGB.r + ', ' + rangeRGB.g + ', ' + rangeRGB.b + ')');
+  d3.select(`#${id}`).append('div')
+    .attr('class', 'tooltip')
+    .style('opacity', '0')
+    .append('p')
+    .attr('class', 'ttvalue');
+  d3.selectAll('rect')
+    .on('mouseenter', function(d){
+      if(g._inEditState) return;
+      var event = d3.event;
+      var x = event.pageX;
+      var y = event.pageY;
+      // var test =d3.select('#' + id);
+      var container = this.parentNode.parentNode.parentNode; // d3.select('#' + id) always gives back the first object's element container, so when a user hover on a 2nd or 3rd or.. bar, its tooltip won't be rendered but the first object tooltip will ,,, DUE to the id not being updated
+      // console.log('d3.select(`#${id}`) ', test._groups[0][0], test);
+      // console.log('nodelist[i] ', nodelist[i]);
+      // // console.log(test._groups[0]);
+      // console.log(id);
+      // console.log(d3.select(container).select('.ttvalue'));
+      // console.log($element);
+      // console.log(d3.select(this));
+
+      // console.log(this.parentNode.parentNode.parentNode);
+      // console.log(nodelist);
+
+      // console.log(layout.qInfo.qId);
+
+      d3.select(container)
+        .select('.ttvalue')
+        .text(d);
+      d3.select(container)
+        .select('.tooltip')
+        .style('left', x + 10 + 'px')
+        .style('top', y - 35 + 'px')
```

Comment:
Other visualisations have their tooltip at their ends or on top. I would suggest to go with the top center same as screenshot #2
<img width="931" alt="screen shot 2019-01-22 at 12 07 22" src="https://user-images.githubusercontent.com/2794781/51532483-7f0fe880-1e40-11e9-8736-b3473277911f.png">
<img width="1438" alt="screen shot 2019-01-22 at 12 22 57" src="https://user-images.githubusercontent.com/2794781/51532484-7f0fe880-1e40-11e9-851d-17dab522c072.png">

Reply:
That case would be applicable for a normal bar chart, but in this case the bullets (bars) in Horizontal view get stacked and having the tool tip on the top won't look so nice since we can't determine how many "stacked bars/bullets" there would be